vllm/compilation/decorators.py

```diff
@@ -1,20 +1,34 @@
-from typing import List, Optional, Union
+import inspect
+from typing import Dict, List, Union
 
 import torch
 
 import vllm.envs as envs
-from vllm.attention import AttentionMetadata
 from vllm.compilation.levels import CompilationLevel
 from vllm.compilation.wrapper import TorchCompileWrapperWithCustomDispatcher
 from vllm.sequence import IntermediateTensors
 from vllm.utils import supports_dynamo
 
 
-def support_compile_llama_style(cls: type):
+def support_torch_compile(dynamic_arg_dims: Dict[str, Union[int, List[int]]]):
+
+    def cls_decorator_helper(cls: type):
+        # helper to pass `dynamic_arg_dims`` to `_support_torch_compile``
+        # to avoid too much indentation for `_support_torch_compile``
+        sig = inspect.signature(cls.forward)
+        for k in dynamic_arg_dims:
+            if k not in sig.parameters:
+                raise ValueError(
+                    f"Argument {k} not found in the forward method of {cls}")
+        return _support_torch_compile(cls, dynamic_arg_dims)
+
+    return cls_decorator_helper
+
+
+def _support_torch_compile(cls: type,
```

Comment:
I think it would be good to have a brief comment describing dynamic_arg_dims and the conventions that are applied for Optional, IntermediateTensors, etc.

Reply:
added in [17a9c2e](https://github.com/vllm-project/vllm/pull/9258/commits/17a9c2eb72ee6e9beb13e4577cdda262e4f9d324).